src/charm.py

```diff
@@ -8,35 +8,208 @@
 """
 
 import logging
+from typing import Optional
 
-from ops.charm import CharmBase, InstallEvent
+from charms.tls_certificates_interface.v2.tls_certificates import (  # type: ignore[import-not-found]  # noqa: E501
+    AllCertificatesInvalidatedEvent,
+    CertificateAvailableEvent,
+    CertificateCreationRequestEvent,
+    CertificateInvalidatedEvent,
+    CertificateRevocationRequestEvent,
+    TLSCertificatesProvidesV2,
+    TLSCertificatesRequiresV2,
+)
+from ops.charm import CharmBase, EventBase
 from ops.main import main
-from ops.model import ActiveStatus
+from ops.model import ActiveStatus, BlockedStatus
 
 logger = logging.getLogger(__name__)
 
+PROVIDES_RELATION_NAME = "certificates-provides"
+REQUIRES_RELATION_NAME = "certificates-requires"
```

Comment:
I agree that things currently are confusing, but even the relation names in `metadata.yaml` could potentially be made clearer. I do not think anything is preventing a user to reverse the endpoints as it is.

It is not an easy problem to solve. See the other comments related to this by @gruyaume.

I think the names you suggested are not necessarily clearer, as no constraints are provided or required, they will be simply applied.

The whole naming things for the relations is the thing that bugs me the most with this charm. I think we need to optimize for users of the charm first, and afterwards align the code.

Any inputs on this @saltiyazan, @DanielArndt, @gruyaume ?

Reply:
I agree with "we need to optimize for users of the charm first, and afterwards align the code". I don't have a strong opinion though, besides we may want to have some docstrings to help developers here to wrap their heads around the whole naming thing.

Reply:
I also agree with optimizing it for users, and also don't really have a strong opinion. Maybe it would be good to break out of the requires/provides naming convention for this case?

Its a weird issue, I'm ok with approving it as is also.

Reply:
I will merge this as-is, and add a task to revisit the naming.